src/detectors/svelte-kit.js

```diff
@@ -0,0 +1,32 @@
+const { hasRequiredDeps, hasRequiredFiles, getYarnOrNPMCommand, scanScripts } = require('./utils/jsdetect')
+
+const FRAMEWORK_PORT = 3000
+
+module.exports = function detector() {
+  // REQUIRED FILES
+  if (!hasRequiredFiles(['package.json'])) return false
+  // REQUIRED DEPS
+  if (!hasRequiredDeps(['@sveltejs/kit'])) return false
+
+  // Everything below now assumes that we are within SvelteKit
+  // For more details refer to https://github.com/sveltejs/kit
+
+  // Please note: Netlify Dev doesn't apply `_redirects` etc. if run via the preferred `svelte-kit dev` command. Since
+  // that is the only benefit of Netlify Dev compared to directly running SvelteKit, we'll simply enforce PROD mode:
+  if (!hasRequiredFiles(['build'])) {
+    throw new Error('SvelteKit detected. Please run `netlify build` first!')
+  }
+
+  const possibleArgsArrs = scanScripts({
+    preferredScriptsArr: ['preview'],
+    preferredCommand: 'svelte-kit preview',
+  })
```

Comment:
```suggestion
  // Everything below now assumes that we are within SvelteKit
  // For more details refer to https://github.com/sveltejs/kit

  const possibleArgsArrs = scanScripts({
    preferredScriptsArr: ['dev'],
    preferredCommand: 'svelte-kit dev',
  })
```

Reply:
Well, as I tried to explain, in my experience Netlify Dev simply won't apply e.g. custom `_redirects` rules if Svelte is started in dev mode. Though, for SvelteKit this would be one of the only benefits of using `netlify dev` instead of `svelte-kit dev`. That's why I made this workaround.

Yes, the user would have to run `svelte-kit build` before each time. But you're right. We should actually investigate first why Netlify Dev doesn't apply `_redirects` in this case...

Reply:
Can you share an example repo? `netlify dev` looks for a redirects file both in the current directory and in the `dist` directory

Reply:
I would expect `preview` to also ignore `_redirects` because you're running the SvelteKit server which knows nothing about them. Just like `dev`, `preview` doesn't use the adapters. You'd have to run the Netlify CLI in order for `_redirects` to do anything (or put code in your `_handle` method that parses and handles the file)

Reply:
@erezrokah Okay, that's interesting. So normally `netlify dev` should work just fine (using `svelte-kit dev`), right? I'll try to investigate later on and give an update...

@benmccann It's not about what SvelteKit does (not) know. It's rather about Netlify Dev not working as it is supposed to.

Reply:
> @erezrokah Okay, that's interesting. So normally `netlify dev` should work just fine (using `svelte-kit dev`), right?

Yes it should

Reply:
I'm dumbfounded as it's (now?) working as expected (with `svelte-kit dev` and all).

@erezrokah I adjusted it accordingly. Additionally setting the `dist` folder correctly.